File: src/ast/BracedExpression.java
package ast;

import java.io.StringWriter;

public class BracedExpression extends BaseNode {
    private BaseNode _element;
    private BaseNode _expression;
    private boolean _isArrayExpression;

    public BracedExpression(BaseNode element, BaseNode expression, boolean isArrayExpression) {
        super(NodeType.BracedExpression);
        _element = element;
        _expression = expression;
        _isArrayExpression = isArrayExpression;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        if (_isArrayExpression) {
            writer.write("[");
            _element.Print(writer);
            writer.write("]");
        } else {
            writer.write(".");
            _element.Print(writer);
        }

        if (!(_expression instanceof BracedExpression) || !(_expression instanceof BracedRangeExpression)) {
            writer.write(" = ");
        }

        _expression.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/BracedExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class BracedExpression(private val element: BaseNode, private val expression: BaseNode, private val isArrayExpression: Boolean) : BaseNode(NodeType.BracedExpression) {
    public override fun printLeft(writer: StringWriter) {
        if (isArrayExpression) {
            writer.write("[")
            element.print(writer)
            writer.write("]")
        } else {
            writer.write(".")
            element.print(writer)
        }
        writer.write(" = ")
        expression.print(writer)
    }

}